src/components/Reports/GoalCalculator/Shared/useReportExpenses/useReportExpenses.tsx

```diff
@@ -0,0 +1,71 @@
+import { useRouter } from 'next/router';
+import { useMemo } from 'react';
+import { PrimaryBudgetCategoryEnum } from 'src/graphql/types.generated';
+import { useAccountListId } from 'src/hooks/useAccountListId';
+import { getQueryParam } from 'src/utils/queryParam';
+import { useGoalCalculationQuery } from '../../SummaryReport/GoalCalculation.generated';
+import { getMinistryExpensesTotal, getPrimaryTotal } from './helpers';
+
+export interface Goal {
+  netMonthlySalary: number;
+  taxesPercentage: number;
+  rothContributionPercentage: number;
+  traditionalContributionPercentage: number;
+  ministryExpenses: MinistryExpenses;
+  ministryExpensesTotal: number;
+}
+
+export interface MinistryExpenses {
+  benefitsCharge: number;
+  primaryCategories: Array<{
+    category: PrimaryBudgetCategoryEnum;
+    label: string;
+    amount: number;
+  }>;
+}
+
+export const useReportExpenses = () => {
+  const accountListId = useAccountListId() ?? '';
+  const { query } = useRouter();
+  const goalCalculationId = getQueryParam(query, 'goalCalculationId') ?? '';
+
+  const { data: goalData, loading } = useGoalCalculationQuery({
```

Comment:
You can get the `goalCalculation` now from `const { goalCalculationResult: { data, loading } } = useGoalCalculator()`.